oidc.go

```diff
@@ -101,8 +102,14 @@ func NewProvider(ctx context.Context, issuer string) (*Provider, error) {
 		return nil, fmt.Errorf("%s: %s", resp.Status, body)
 	}
 	defer resp.Body.Close()
+
 	var p providerJSON
 	if err := json.Unmarshal(body, &p); err != nil {
+		var isJSON bool
+		content, _, _ := mime.ParseMediaType(r.Header.Get("Content-Type"))
```

Comment:
I don't think so, we only get an error if it cannot parse the media type, which means `content` won't be anything we're explicitly looking for, it will just be "". Since we're providing our own error message, I don't think we need the media type parsing error.